### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Kingno3l**
### Summary of Changes

This pull request implements the following features and relationships across various classes:

#### Classroom Class
- Created the `Classroom` class with an `@label` instance variable.
- Added setter and getter methods using `attr_accessor`.
- Established a has-many/belongs-to relationship between `Classroom` and `Student`:
  - A classroom has many students.
  - A student belongs to a classroom.
  - Implemented functionality:
    - When adding a student to a classroom, it automatically sets the classroom for the student.
    - When setting the classroom for a student, it also adds the student to the classroom's students.

#### Book Class
- Developed the `Book` class with `@title` and `@author` instance variables.
- Included setter and getter methods utilizing `attr_accessor`.

#### Rental Class
- Constructed the `Rental` class with an `@date` instance variable.
- Implemented setter and getter methods for `@date` using `attr_accessor`.
- Created a many-to-many (has-many-through) relationship between `Person` and `Book` using the intermediate class `Rental`:
  - A book has many rentals.
  - A rental belongs to a book.
  - A person has many rentals.
  - A rental belongs to a person.
- Modified the constructor of `Rental` to set `Book` and `Person` instances.

### Details
The changes introduce and link classes to fulfill the specified project requirements:
- Added `Classroom`, `Book`, and `Rental` classes.
- Established relationships between `Classroom` and `Student`, as well as `Person` and `Book` using `Rental`.

### Notes
The modifications aim to achieve a structured relationship between entities as outlined in the project requirements, enabling proper associations between classrooms, students, books, and rentals.

---

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
```
Comment (prolajumokeoni) [inline classroom.rb]: ### OPTIONAL

- [ ] Kindly consider changing `attr_accessor :students` to `attr_reader :students` for better control. This modification will grant access to the students' information without allowing direct changes 👍🏿

File: student.rb [hunk 2]
```diff
@@ -1,9 +1,17 @@
 class Student < Person
-  attr_reader :classroom
+  attr_accessor :classroom
```
Comment (prolajumokeoni) [inline student.rb]: ###  OPTIONAL 
- [ ] Kindly consider updating `attr_accessor :classroom` to `attr_reader :classroom` for improved control. This adjustment will allow access to the classroom information without permitting direct modifications 👍🏿

**Review by prolajumokeoni — APPROVED** [review]
Hi @Kingno3l 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Associations  set up correctly between classes✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (Kingno3l) [reply classroom.rb]: attr_reader :label, :students
this is what i have now, is it correct? @prolajumokeoni

File: student.rb (hunk 2 above)
Comment (Kingno3l) [reply student.rb]: Also this is what I now have on line 2
  attr_reader :classroom


is it correct ma? @prolajumokeoni

File: classroom.rb (hunk 1 above)
Comment (prolajumokeoni) [reply classroom.rb]: Yes @Kingno3l well-done implementing the changes 👏🏿

**Pull request merged**